### PR TITLE
Add Storage Saving Functionality for Sleep Diary Data

### PR DESCRIPTION
**Pull request by itsDZhang**
#### Link to the issue: 
https://github.com/de-souza-capstone-2020/Mars/issues/55

#### Description of Pull Request:

Saves data to storage

#### Test Plan:

- [ ] Fill in modal data
- [ ] See your console and the data you inputted should be logged

### CODE REVIEW

File: screens/sleepDiary.js [hunk 1]
```diff
@@ -1,157 +1,327 @@
...
     sleepTime: new Date(),
-    sleepQuality: ''
-  }
+    attemptToSleepTime: new Date(),
+    durationTillSleep: "",
+    numTimesWakeUp: "",
+    durationTotalWakeUp: "",
+    wakeUpTime: new Date(),
+    leaveBedTime: new Date(),
+    didNap: false,
```
Comment (mllemango) [inline screens/sleepDiary.js]: `didNap: ""` since its a string

File: screens/sleepDiary.js [hunk 2]
```diff
@@ -1,157 +1,327 @@
...
+
+            <View style={styles.row}>
+              <View style={styles.question}>
+                <Text style={styles.text}>
+                  Did you nap today? If so, when and for how long?
+                </Text>
+              </View>
+              <View style={styles.answer}>
+                <Picker
+                  selectedValue={"no"}
```
Comment (mllemango) [inline screens/sleepDiary.js]: `selectedValue={didNap}`

File: screens/sleepDiary.js [hunk 3]
```diff
@@ -1,157 +1,327 @@
...
+                    this.setState({ didNap: itemValue })
+                  }
+                >
+                  <Picker.Item label="No" value="no" />
+                  <Picker.Item label="Yes" value="yes" />
+                </Picker>
+              </View>
+            </View>
+
+            {didNap && (
```
Comment (mllemango) [inline screens/sleepDiary.js]: `didNap === "yes"`

File: screens/sleepDiary.js [hunk 4]
```diff
@@ -1,157 +1,327 @@
...
+                    onDateChange={napTime => this.setState({ napTime })}
+                    mode={"time"}
+                    fadeToColor={"none"}
+                    textColor={"#000000"}
+                    style={{ height: 50, width: 120 }}
+                  />
+                </View>
+              </View>
+            )}
+            {didNap && (
```
Comment (mllemango) [inline screens/sleepDiary.js]: same as above

**Comment by itsDZhang** [comment]
@mllemango resolved 😄

**Pull request merged**